### ISSUE TITLE
DRep "expiry" is incremented even for inactive DReps

### ISSUE THREAD

**Issue by mkoura** [IntersectMBO/cardano-ledger#4346]
The DRep "expiry" is incremented when DRep votes or when there is no proposal to vote for. It would make sense to increment the expiry only for DReps that are not expired, so the "expiry" value can be used for finding out when the DRep became inactive. As it is now, the "expiry" value is incremented also for inactive DReps.

E.g. DRep state in epoch 12:

```json
[
  [
    {
      "keyHash": "00318f9ab74f528241094f51f059d916ed4fbadc748ccffb69168ed9"
    },
    {
      "anchor": null,
      "deposit": 2000000,
      "expiry": 11
    }
  ]
]
```

Current epoch is > "expiry", so the DRep is inactive. In epoch 14 the status is
```json
[
  [
    {
      "keyHash": "00318f9ab74f528241094f51f059d916ed4fbadc748ccffb69168ed9"
    },
    {
      "anchor": null,
      "deposit": 2000000,
      "expiry": 12
    }
  ]
]
```
meaning the "expiry" was incremented (there was one proposal that the DRep didn't vote on, so the "expiry" was incremented just by one).

**Comment by lehins** [comment]
Turns out there is a subtle bug, since the counter should not have been incremented when there is at least one proposal to vote on. Here is the issue #4347 

That being said, the counter will be incremented when there are no proposals. The interesting fact is that we do not increment a value for every DRep, we have a separate counter incremented for all DReps, regardless if they have expired or not. That does not mean though that we cannot implement this feature on the ledger query side here:

https://github.com/IntersectMBO/cardano-ledger/blob/760ea5ae5de8ba0ee0568252e36e57ef4737ed63/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L138-L147

All we need to do is do not change the `drepExpiry` for those DReps that have already expired, which can all be done on this line:
https://github.com/IntersectMBO/cardano-ledger/blob/760ea5ae5de8ba0ee0568252e36e57ef4737ed63/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L144

We should also add a test for this functionality in the `cardano-ledger-api` test suite for this.

**Comment by lehins** [comment]
Fixed in #4358
